src/Illuminate/Foundation/Testing/Concerns/InteractsWithDatabase.php

```diff
@@ -96,4 +98,43 @@ public function seed($class = 'DatabaseSeeder')
 
         return $this;
     }
+
+    /**
+     * Start counting executed database queries.
+     *
+     * @return $this
+     */
+    public function startCountingDatabaseQueries()
```

Comment:
IMHO this shouldn't be necessary to be called explicitly. The setup for a trait should happen automagically when a trait gets added, see how `\Illuminate\Foundation\Testing\TestCase::setUpTraits` handles this with other traits.

Reply:
I thought about this, but if it is automatically starts counting, then it also counts all the queries your factories make when setting up the test data. Explicitly calling it ensures that you only count queries from production code.

Reply:
Sorry for not being clear, I just meant the trait setup initialization part.

Yes, the ability to reset the counters must be available.

Btw, it's technically not wrong to call this multiple times in a test, thus I literally would name it something including "reset", e.g. "resetCountingDatabaseQueries"…